File: src/main/kotlin/com/nazarethlabs/codex/service/note/ChangeNoteColorService.kt
package com.nazarethlabs.codex.service.note

import com.nazarethlabs.codex.dto.Note
import com.nazarethlabs.codex.enum.NoteColorEnum
import com.nazarethlabs.codex.repository.NoteStorageRepository

class ChangeNoteColorService {
    private val noteStorage = NoteStorageRepository.getInstance()

    fun changeColor(
        note: Note,
        color: NoteColorEnum,
    ) {
        noteStorage.changeColor(note.id, color)
    }
}

File: src/service/ChangeNoteColorService.ts
import { Note } from '../dto/Note';
import { NoteColorEnum } from '../enum/NoteColorEnum';
import { NoteRepository } from '../repository/NoteRepository';
import { DialogHelper } from '../helper/DialogHelper';

interface ColorOption {
  label: string;
  value: NoteColorEnum;
}

export class ChangeNoteColorService {
  private readonly repository: NoteRepository;

  constructor(repository: NoteRepository) {
    this.repository = repository;
  }

  async changeColor(note: Note): Promise<void> {
    const options: ColorOption[] = [
      { label: 'None', value: NoteColorEnum.NONE },
      { label: 'Blue', value: NoteColorEnum.BLUE },
      { label: 'Green', value: NoteColorEnum.GREEN },
      { label: 'Yellow', value: NoteColorEnum.YELLOW },
      { label: 'Orange', value: NoteColorEnum.ORANGE },
      { label: 'Pink', value: NoteColorEnum.PINK },
      { label: 'Red', value: NoteColorEnum.RED },
      { label: 'Purple', value: NoteColorEnum.PURPLE },
    ];

    const selected = await DialogHelper.showQuickPick(
      options.map((o) => ({ label: o.label, description: o.value, value: o.value })),
      { placeHolder: 'Select note color' },
    );

    if (!selected) {
      return;
    }

    this.repository.changeColor(note.id, selected.value);
  }
}
